zk_toolbox/crates/zk_inception/src/commands/ecosystem/create.rs

```diff
@@ -61,7 +66,14 @@ fn create(args: EcosystemCreateArgs, shell: &Shell) -> anyhow::Result<()> {
         spinner.finish();
         link_to_code
     } else {
-        let path = PathBuf::from_str(&args.link_to_code)?;
+        let mut path = PathBuf::from_str(&args.link_to_code)?;
+        if let Err(err) = check_link_to_code(shell, &path) {
+            logger::warn(err);
+            if !PromptConfirm::new(MSG_CONFIRM_STILL_USE_FOLDER).ask() {
+                path = pick_new_link_to_code(shell)?;
+            }
+        }
```

Comment:
could we make a while here?

Reply:
https://github.com/matter-labs/zksync-era/pull/2447/files#diff-033159fff9cca087b54d0391a4b7631a96a20149f343958d70750e42a36d1641R157

`pick_new_link_to_code` is recursive already but I can replace for while and move it to that part of the code. What do you think?

Reply:
oh, I missed it. it's ok